vllm/model_executor/models/mamba_cache.py

```diff
@@ -0,0 +1,222 @@
+from typing import Dict, List, Optional
+
+import torch
+
+from vllm.attention.backends.abstract import AttentionMetadata
+
+
+class MambaCacheManager:
```

Comment:
This is a nice abstraction! Agree that moving into BlockManager eventually makes sense